code/game/objects/items/devices/aicard.dm

```diff
@@ -31,61 +31,64 @@
 		to_chat(user, "<b>ERROR ERROR ERROR</b>")
 
 /obj/item/aicard/attack_self(mob/user)
-
 	ui_interact(user)
 
-/obj/item/aicard/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = inventory_state)
-	var/data[0]
+/obj/item/aicard/ui_interact(mob/user, datum/topic_state/state = inventory_state)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+
+	if (!ui)
+		ui = new(user, src, "devices-aicard", 600, 400, "[name]", state = state)
+		ui.auto_update_content = TRUE
+
+	ui.open()
+
+/obj/item/aicard/vueui_data_change(list/data, mob/user, datum/vueui/ui)
+	data = list()
```

Comment:
Please also add all initial values for fields, as during data sending all fields must be present at all time, because rectivity caviats - https://vuejs.org/v2/guide/reactivity.html